internal/jwks/handle_update_all.go

```diff
@@ -29,7 +30,9 @@ const jwksLockID = "jwks-import"
 
 func (s *Server) handleUpdateAll() http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		ctx := r.Context()
+		ctx, cancel := context.WithTimeout(r.Context(), s.config.MaxRuntime)
```

Comment:
I think you need to handle the context error somewhere else in this body then?

Reply:
this is to set up the semaphore so it will have a maximum wait time on the acquire